tests/test_data_types.py

```diff
@@ -0,0 +1,146 @@
+def check_data_type_interface(data_type):
+    assert callable(data_type.serialize)
+    assert callable(data_type.deserialize)
+    assert callable(data_type.to_dict)
+
+# BASIC TESTS FOR ALL DATA TYPES -----------------------------------------------
+def test_data_type_interface_any():
+    check_data_type_interface(any)
+
+def test_data_type_interface_array():
+    check_data_type_interface(array)
+
+def test_data_type_interface_image():
+    check_data_type_interface(image)
+
+def test_data_type_interface_vector():
+    check_data_type_interface(vector)
+
+def test_data_type_interface_category():
+    check_data_type_interface(category)
+
+def test_data_type_interface_number():
+    check_data_type_interface(number)
+
+def test_data_type_interface_text():
+    check_data_type_interface(text)
+
+def test_data_type_interface_file():
+    check_data_type_interface(file)
+
+# TEXT -------------------------------------------------------------------------
+def test_text_to_dict():
+    default = 'Some default text'
+    txt = text(default=default, min_length=1, max_length=20)
+    obj = txt.to_dict()
+    assert obj['type'] == 'text'
+    assert obj['name'] == 'text'
+    assert obj['default'] == default
+    assert obj['minLength'] == 1
+    assert obj['maxLength'] == 20
+
+def test_text_serialization():
+    txt = text()
+    assert txt.serialize(512) == '512'
+
+def test_text_deserialize():
+    txt = text()
+    assert txt.deserialize('512') == '512'
+
+# NUMBER -----------------------------------------------------------------------
+def test_number_to_dict():
+    default = 42
+    num = number(default=default, min=10, max=100)
+    obj = num.to_dict()
+    assert obj['type'] == 'number'
+    assert obj['name'] == 'number'
+    assert obj['default'] == default
+    assert obj['min'] == 10
+    assert obj['max'] == 100
+    assert obj['step'] == 1
+
+def test_number_serialization():
+    assert 1 == number().serialize(1)
+    assert 1.1 == number().serialize(1.1)
+
+def test_number_deserialize():
+    assert 1 == number().deserialize(1)
+    assert 1.1 == number().deserialize(1.1)
+
+def test_number_deserialize_numpy_scalar():
+    assert 10 == number().deserialize(np.array(10))
+
+def test_number_serialize_numpy_scalar():
+    assert 10 == number().serialize(np.array(10))
+
+# ARRAY ------------------------------------------------------------------------
+def test_array_to_dict():
+    arr = array(item_type=text, min_length=5, max_length=10)
+    obj = arr.to_dict()
+    assert obj['name'] == 'text_array'
+    assert obj['type'] == 'array'
+    assert obj['itemType'] == text().to_dict()
+    assert obj['minLength'] == 5
+    assert obj['maxLength'] == 10
+
+def test_array_no_item_type():
+    with pytest.raises(MissingArgumentError):
+        arr = array()
+
+def test_array_serialization():
+    expect = ['10', '100', '1000']
+    assert expect == array(item_type=text).serialize([10, 100, 1000])
```

Comment:
Even though this works for now (since we have no output validation) it's not the ideal test case as `[10, 100, 1000]` isn't an `array(text)`. Might be better to use an `item_type=vector`, and then test that `serialize` converts the value from `list(np.array) -> list(list)` properly?

Reply:
Ah, see I was actually testing the string casting behavior of `text().serialize()`, which I assumed was intentional: https://github.com/runwayml/model-sdk/blob/72bcee05f616642140f389f1f2fddf3ba6ea3f86/runway/data_types.py#L367-L368

But I see that point. Maybe I'll leave this one, so that if this behavior changes in the future the test will at least fail (and we'll know about it), and then also add your recommended test.